app/models/embed/purl/resource_file.rb

```diff
@@ -43,6 +43,14 @@ def thumbnail?
         Settings.resource_types_that_contain_thumbnails.include?(resource.type)
       end
 
+      def vtt
+        resource.files.find(&:vtt?)
+      end
+
+      def vtt?
+        mimetype == 'text/plain' && title.end_with?('.vtt')
```

Comment:
Shouldn't this be `text/vtt`? https://developer.mozilla.org/en-US/docs/Web/API/WebVTT_API#webvtt_files

Reply:
I think that's right @jcoyne. I think it might be coded in the example XML that way? Perhaps it should allow for both?

Reply:
I think the example XML is just bad data.

Reply:
It looks there are 45 text/vtt files at the moment: https://argo.stanford.edu/catalog?all=true&f[content_file_mimetypes_ssim][]=text%2Fvtt 

I guess I'd need to run a report to see if there are many/any `.vtt` files that need the correct media type.

Reply:
Now I'm wondering if it should ignore the extension altogether and only key of the media type?

Reply:
Yes, I don't think the extension is something we care about.

Reply:
It's not exactly bad data. vtt is a text file. What we need is more precise identification of text files. Infrastructure will need to build that. We have similar identification problems with lots of text file types. The more specifc the consuming applications need mimetypes, the more precisely the accessioning apps need to identify them.

We don't want a situation where a person has to manually correct mimetypes after each accession. If you ignore the extension here, I think the most likely the path to identifying VTT files as text/vtt will be to use the extension during accessioning. Which may be the best way to do it, but there's both trade-offs and cross-team implications to whatever strategy we choose.

Reply:
That's an important point @andrewjbtw -- that would suck to have to in and manually fix anything with a transcript.  Does techmd do the mediatype detection for us at the moment? Do we need a ticket in there, and perhaps a data remediation ticket as well?

Reply:
Improvements to mime-type detection should go in https://github.com/sul-dlss/assembly-objectfile/blob/229520e3c590dc3e8c83157c5d62c00cd3e53eb7/lib/assembly/object_file.rb#L103. 

It does have a problem with this type as it prefers the value returned from `file` before the value from `extension`

```ruby
file = Assembly::ObjectFile.new('foo.vtt')
file.send :exif_mimetype
=> nil
file.send :file_mimetype
=> "text/plain"
file.send :extension_mimetype
=> "text/vtt"
```

Reply:
There are a couple of different places where mime-types are determined. I think `sdr-api` does it as well. Maybe counterintuitively, the mime-type in the structural metadata is generated and stored separately from the mime-type in the technical metadata, which has been the case going back to the Fedora era. The access systems don't use the technical metadata.

Reply:
FWIW, I just deposited a VTT file using the SDR API in stage, and it appears both sdr-api and the techmd service applied the correct MIME type: https://argo-stage.stanford.edu/view/druid:pc587kh4617

assembly-objectfile may be the one spot needing attention here as @jcoyne shared above.

Reply:
I was wrong about the above: @edsu already patched `assembly-objectfile` such that is correctly picks the VTT mime type, and that's in the latest version of the gem.